rust/arrow/src/csv/reader.rs

```diff
@@ -99,7 +99,27 @@ fn infer_field_schema(string: &str) -> DataType {
 /// If `max_read_records` is not set, the whole file is read to infer its schema.
 ///
 /// Return infered schema and number of records used for inference.
-fn infer_file_schema<R: Read + Seek>(
+pub fn infer_file_schema<R: Read + Seek>(
+    reader: &mut R,
+    delimiter: u8,
+    max_read_records: Option<usize>,
+    has_header: bool,
+) -> Result<(Schema, usize)> {
+    let (schema, records_count) =
+        infer_schema_from_reader(reader, delimiter, max_read_records, has_header)?;
+    // return the reader seek back to the start
+    reader.seek(SeekFrom::Start(0))?;
+
+    Ok((schema, records_count))
+}
+
+/// Infer schema of CSV records provided by struct that implements `Read` trait.
+///
+/// `max_read_records` controlling the maximum number of records to read. If `max_read_records` is
+/// not set, all records are read to infer the schema.
+///
+/// Return infered schema and number of records used for inference.
+pub fn infer_schema_from_reader<R: Read>(
```

Comment:
```suggestion
pub fn infer_reader_schema<R: Read>(
```

To mirror the name of `infer_file_schema`? I don't feel strongly about this but wanted to suggest it

Reply:
:+1: